core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3202,6 +3215,60 @@ object Stream extends StreamLowPriority {
   def fromBlockingIterator[F[_]]: PartiallyAppliedFromIterator[F] =
     new PartiallyAppliedFromIterator(blocking = true)
 
+  /** Returns a stream of elements from the supplied queue.
+    *
+    * The stream terminates upon dequeuing a `None`.
+    *
+    * All elements that are available, up to the specified limit,
+    * are dequeued and emitted as a single chunk.
+    */
+  def fromQueueNoneTerminated[F[_]: Functor, A](
```

Comment:
Assuming cats-effect gets an `Dequeue` type, maybe this should be named `dequeueNoneTerminated` instead of `fromQueueNoneTerminated`.

Reply:
I think it already has a `Dequeue` type

Reply:
Eek, namespace clash -- I mean a trait that has `take` and `tryTake` but not `offer` or `tryOffer`